docs/user-guide/model.rst

```diff
@@ -0,0 +1,66 @@
+.. _model:
+
+Defining the model for a form
+-----------------------------
+
+The underlying schema for a ``c2cgeoform`` form is defined as SQLAlchemy
+model. A simple definition is shown below:
+
+.. code-block:: python
+
+   from sqlalchemy import (Column, Integer, Text)
+   import deform
+   from uuid import uuid4
+
+   from c2cgeoform.models import Base
+
+
+   class Comment(Base):
+       __tablename__ = 'comments'
+       __colanderalchemy_config__ = {
+           'title': 'A very simple form'
+       }
+
+       id = Column(Integer, primary_key=True, info={
+           'colanderalchemy': {
+               'widget': deform.widget.HiddenWidget()
+           }})
+
+       hash = Column(Text, unique=True, default=lambda: str(uuid4(), info={
+           'colanderalchemy': {
+               'widget': HiddenWidget()
+           }})
+
+       name = Column(Text, nullable=False, info={
+           'colanderalchemy': {
+               'title': 'Name'
+           }})
+
+       comment = Column(Text, nullable=True, info={
+           'colanderalchemy': {
+               'title': 'Comment',
+               'widget': deform.widget.TextAreaWidget(rows=3),
+           }})
+
+This SQLAlchemy model is enriched with properties for
```

Comment:
This SQLAlchemy model is enriched with properties for
`ColanderAlchemy`_, for example to set a title for a field, use a
specific Deform `widget`_ or use a Colander `validator`_.